Crashlytics/Crashlytics/Components/FIRCLSProcess.c

```diff
@@ -485,6 +485,9 @@ bool FIRCLSProcessRecordAllThreads(FIRCLSProcess *process, FIRCLSFile *file) {
 
     FIRCLSSDKLogInfo("recording thread %d data\n", i);
     if (!FIRCLSProcessRecordThread(process, thread, file)) {
+      FIRCLSFileWriteArrayEnd(file);
```

Comment:
Just one suggestion here: could you add an error SDK Log Error, eg:

`FIRCLSSDKLogError("Failed to record thread state. Closing threads JSON to prevent malformed crash report.")`

Reply:
Sure, since watchOS doesn't support thread states, should I wrap that if in a supported macro? Otherwise it will always log that error message for watchOS.

Reply:
I think it's ok to leave as-is. This will remind folks in the future that this is something we can improve upon 👍